formats.go

```diff
@@ -22,6 +22,28 @@ type FrameSize struct {
 	StepHeight uint32
 }
 
+// Represents framerate supported by a webcam
+type FrameRate struct {
```

Comment:
how does one know which values to take it out in a way that prevents misuse?

Reply:
Added comment about discrete returns. For stepwise returns, the user can expand based on min/max/step to retrieve all supported FPS.

Reply:
what about an emum type to help with this? that way you dont need to figure out the logic from the comment

Reply:
There is an enum for handling return type:

```
const (
	V4L2_FRMIVAL_TYPE_DISCRETE   uint32 = 1
	V4L2_FRMIVAL_TYPE_CONTINUOUS uint32 = 2
	V4L2_FRMIVAL_TYPE_STEPWISE   uint32 = 3
)
```

I do not necessarily like how we have one data structure `FrameRate` for handling both Discrete and Stepwise returns because the user has to check step size and handle accordingly. However, this is the way that FrameSize currently works.

Maybe we should just return an array of fracts `[]v4l2_fract`.

Reply:
I guess if you _really_ wanted to you could return the type so users wouldn't have to check the steps to figure out the type. That would also future-proof this a bit because if we do decide to support `V4L2_FRMIVAL_TYPE_CONTINUOUS` we might actually want those types exposed. However, this behavior is consistent with what's already here and I'd be ok with this leaving this just like it is and punt returning types until we actually do support `V4L2_FRMIVAL_TYPE_CONTINUOUS`.

Reply:
Fine with me